extras/indexing/src/main/java/mvm/rya/sail/config/RyaSailFactory.java

```diff
@@ -55,91 +52,98 @@
-            final int port = Integer.parseInt(((MongoDBRdfConfiguration)conf).getMongoPort());
-
-            MongoClient client = null;
-            try {
-                client = new MongoClient(
-                    new ServerAddress(hostname, port),
-                    Lists.newArrayList(creds));
-                ryaDetailsRepo = new MongoRyaInstanceDetailsRepository(client, mongoConf.getCollectionName());
-                RyaDetailsToConfiguration.addRyaDetailsToConfiguration(ryaDetailsRepo.getRyaInstanceDetails(), mongoConf);
-            } catch(final RyaDetailsRepositoryException e) {
-                Log.info("Instance does not have a rya details collection, skipping.");
-            } catch (final UnknownHostException ue) {
-                throw new RyaDAOException("Unable to connect to mongo at the configured location.", ue);
-            } finally {
-                ConfigUtils.setIndexers(conf);
-
-                if(client != null) {
-                    crdfdao = new MongoDBRyaDAO(mongoConf, client);
-                } else {
-                    try {
-                        crdfdao = new MongoDBRyaDAO(mongoConf);
-                    } catch (NumberFormatException | UnknownHostException e) {
-                        e.printStackTrace();
-                    }
-                }
-                crdfdao.init();
-
-                conf.setDisplayQueryPlan(true);
-                store.setRyaDAO(crdfdao);
-            }
+            final MongoClient client = updateMongoConfig((MongoDBRdfConfiguration) conf, user, pswd, instance);
+            dao = getMongoDAO((MongoDBRdfConfiguration)conf, client);
         } else {
             conf = new AccumuloRdfConfiguration(config);
-            try {
-                final PasswordToken pswdToken = new PasswordToken(pswd);
-                final Instance accInst = ConfigUtils.getInstance(conf);
-                ryaDetailsRepo = new AccumuloRyaInstanceDetailsRepository(accInst.getConnector(user, pswdToken), instance);
-                RyaDetailsToConfiguration.addRyaDetailsToConfiguration(ryaDetailsRepo.getRyaInstanceDetails(), conf);
-            } catch(final RyaDetailsRepositoryException e) {
-                Log.info("Instance does not have a rya details collection, skipping.");
-            } finally {
-                final Connector connector = ConfigUtils.getConnector(config);
-                crdfdao = new AccumuloRyaDAO();
-                ((AccumuloRyaDAO)crdfdao).setConnector(connector);
-
-                conf.setTablePrefix(config.get(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX)); // sets
-                // TablePrefixLayoutStrategy
-                ConfigUtils.setIndexers(conf);
-                conf.setDisplayQueryPlan(true);
-
-                crdfdao.setConf(conf);
-                crdfdao.init();
-                store.setRyaDAO(crdfdao);
-            }
+            updateAccumuloConfig((AccumuloRdfConfiguration) conf, user, pswd, instance);
```

Comment:
You're using the wrong value here too. AccumuloRyaInstanceDetailsRepository requires the Rya instance name, not the Accumulo instance name. I think you're confused about what the 'ConfigUtils.CLOUDBASE_INSTANCE' field indicates.